frontend/src/components/JiraDashboard/components/WidgetJira.js

```diff
@@ -0,0 +1,198 @@
+import { Button, SIZE } from "baseui/button";
+import { Select } from "baseui/select";
+
+import Table from "../../Table/Table";
+import useFilters from "../hooks/useFilters";
+import useEmployeeFilter from "../hooks/useEmployeeFilter";
+import useGetJiraData from "../hooks/useGetJiraData";
+
+const columnTitles = ["Type", "Key", "Summary", "Status", "Priority"];
+const title = "All Issues";
+
+const Widgetjira = () => {
+    const [jqlQuery, setJqlQuery] = useState(""); //jqlQuery that we pass as prop to table
+    const [inputValue, setInputValue] = useState(""); //change in Input JQL query
+    const [filterValue, setFilterValue] = useState([]); //change in Filter dropdown
+    const [employeeFilterValue, setEmployeeFilterValue] = useState([]); //change in Employee Filter dropdown
+    const [basicMode, setBasicMode] = useState(false);
+
+    const { filters } = useFilters();
+    const { employeeDetails } = useEmployeeFilter();
+    const { data, pageNumber, totalPages, setPageNumber, loading, errMessage } =
+        useGetJiraData(jqlQuery);
+
+    const handleInputChange = useCallback(
+        (event) => {
+            setInputValue(event.target.value);
+        },
+        [inputValue]
+    );
+
+    const handleClick = useCallback(() => {
+        setJqlQuery(inputValue);
+    }, [inputValue]);
+
+    const handleFilter = useCallback(
+        (params) => {
+            setFilterValue(params.value);
+            if (params.value.length > 0 && employeeFilterValue.length > 0) {
+                setJqlQuery(
+                    `filter=${params.value[0].id} AND assignee in ("${employeeFilterValue[0].id}")`
+                );
+            } else if (params.value.length > 0) {
+                setJqlQuery(`filter=${params.value[0].id}`);
+            } else if (employeeFilterValue.length > 0) {
+                setJqlQuery(`assignee in ("${employeeFilterValue[0].id}`);
+            } else {
+                setJqlQuery("");
+            }
+        },
+        [employeeFilterValue]
+    );
+
+    const handleEmployeeFilter = useCallback(
+        (params) => {
+            setEmployeeFilterValue(params.value);
+            if (params.value.length > 0 && filterValue.length > 0) {
+                setJqlQuery(
+                    `filter=${filterValue[0].id} AND assignee in ("${params.value[0].id}")`
+                );
+            } else if (params.value.length > 0) {
+                setJqlQuery(`assignee in ("${params.value[0].id}")`);
+            } else if (filterValue.length > 0) {
+                setJqlQuery(`filter=${filterValue[0].id}`);
+            } else {
+                setJqlQuery("");
+            }
+        },
+        [filterValue]
+    );
+
+    const handleSwitch = useCallback(() => {
+        setBasicMode((prevMode) => !prevMode);
+    }, [basicMode]);
+    return (
+        <>
+            <div className="jiraWid">
+                <div className="jiraWid__search">
+                    {basicMode === false ? (
```

Comment:
no need to write basicMode === false. you can also write !basicMode ? x : y or basicMode ? y : x